### PR TITLE
ec2_vpc_endpoint use module_utils helper for tagging

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

- `ec2_vpc_endpoint` use module_utils helper for tag management

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ec2_vpc_endpoint

### CODE REVIEW

**Review by tremble — CHANGES_REQUESTED** [review]
Generally looks good, but as with nat gateway it's now possible to pass TagSpecifications during creation (since botocore 1.14.11)

**Comment by alinabuzachis** [comment]
> Generally looks good, but as with nat gateway it's now possible to pass TagSpecifications during creation (since botocore 1.14.11)

@tremble Thanks. I didn't check that, sorry. I'll update accordingly.

File: plugins/modules/ec2_vpc_endpoint.py [hunk 1]
```diff
@@ -358,6 +327,9 @@ def create_vpc_endpoint(client, module):
 
     if policy:
         params['PolicyDocument'] = json.dumps(policy)
+
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint.py:330]:
```suggestion

```

**Review by tremble — APPROVED** [review]

File: plugins/modules/ec2_vpc_endpoint.py [hunk 2]
```diff
@@ -213,13 +213,14 @@
...
 from ansible_collections.amazon.aws.plugins.module_utils.core import normalize_boto3_result
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_tag_list
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import compare_aws_tags
-
 from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
 from ansible_collections.amazon.aws.plugins.module_utils.waiters import get_waiter
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ensure_ec2_tags
```
Comment (tremble) [inline plugins/modules/ec2_vpc_endpoint.py:218]:
```suggestion
from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ensure_ec2_tags
from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_tag_list
```

**Pull request merged**